source/tests/system/nidcpower_driver_api_tests.cpp

```diff
@@ -447,6 +474,28 @@ TEST_F(NiDCPowerDriverApiTest, ConfigureOutputFunctionAndCurrentLevel_Configures
   EXPECT_EQ(expected_output_function_value, actual_output_function_value);
 }
 
+TEST_F(NiDCPowerDriverApiTest, SetMeasureWhenAndInitiate_FetchMultiple_ReturnsResultOfExpectedSize)
+{
+  const char* channel_name = "0";
+  const int expected_count = 20;
```

Comment:
How do you know the expected channel count is 20 when you're not in control of exactly which piece of DC power hardware will be present when this test is run?

The system tests are there just to make sure that you can transfer *some* data to/from the driver from the gRPC server. In order to test correctness of the data being returned or any corner cases, you'll need to write an integration test using the mock library.

Reply:
In this particular case, I believe unit test would be sufficient too. Only "outside" interaction [FetchMulitple](https://github.com/doshirohan/grpc-device/blob/features/ni-dcpower/generated/nidcpower/nidcpower_service.cpp#L2171) has is with `SessionRepository` and our [ni-fake unit test ](https://github.com/doshirohan/grpc-device/blob/features/ni-dcpower/source/tests/unit/ni_fake_service_tests.cpp#L45)seems to be using SessionRespository instance as well. Thoughts?

Reply:
Actually now that I think about it, the unit test added against ni-fake metadata does take care of validating that API returns correct values (conversion from ViBoolean[] to bool[] worked as expected). That's what essentially I was trying to do with expected_count check here. I will update this system test just to check if API returned success and let existing unit test worry about validity of data.

Reply:
I have removed assumptions regarding size and we will rely on unit test [NiFakeService_BoolArrayOutputFunction_CallsBoolArrayOutputFunction](https://github.com/doshirohan/grpc-device/blob/features/ni-dcpower/source/tests/unit/ni_fake_service_tests.cpp#L501) on nifake to verify integrity of data.